tests/test_update_bed.py

```diff
@@ -0,0 +1,235 @@
+    bed2add_path.write_text(mock_bed_content)
+
+    fourth_col_default = "default_annot"
+    append_to_bed(out_bed_path, bed2add_path, fourth_col_default)
+    result = out_bed_path.read_text()
+
+    assert result == expected_output, f"Output was: {result}"
+
+
+# FIXME: Investigate the CLNSIG, CLNSIGCONF, CLNSIGINCL combinations in real ClinVar data
+def test_read_clinvar(tmp_path: Path):
+
+    mock_vcf_content = """\
+##fileformat=VCFv4.2
+##source=ClinVar
+#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO
+chr1\t12345\t.\tA\tG\t.\t.\tCLNSIG=Pathogenic;CLNSIGCONF=Conflicting_interpretations_of_pathogenicity;CLNSIGINCL=athogenic
+chr1\t54321\t.\tC\tT\t.\t.\tCLNSIG=Benign
+chr2\t12345\t.\tG\tA\t.\t.\tCLNSIG=Likely_pathogenic
+"""
+
+    clinvar_vcf_path = tmp_path / "tmp.vcf"
+    clinvar_vcf_path.write_text(mock_vcf_content)
+
+    clinvar_variants, benign_clinvar_variants = read_clinvar(clinvar_vcf_path)
+
+    expected_clinvar_reasons = {
+        "chr1:12345_A_G": "Pathogenic",
+        "chr2:12345_G_A": "Likely_pathogenic",
+    }
+    expected_benign_clinvar_reasons = {"chr1:54321_C_T": "Undefined"}
+
+    for key, variant in clinvar_variants.items():
+        assert key in expected_clinvar_reasons
+        assert expected_clinvar_reasons[key] == variant.reason
+
+    for key, variant in benign_clinvar_variants.items():
+        assert key in expected_benign_clinvar_reasons
+        assert expected_benign_clinvar_reasons[key] == variant.reason
+
+    # print(clinvar_variants)
+    # print(benign_clinvar_variants)
+
+
+def test_compare_clinvar(tmp_path: Path):
+
+    new_clinvar = {
+        "chr1:1000_A_G": ClinVarVariant(
+            "chr1", 1000, "A", "G", {"CLDN": "CLDN", "CLNACC": "CLNACC"}
+        ),
+        "chr2:3000_C_T": ClinVarVariant(
+            "chr2", 3000, "C", "T", {"CLDN": "CLDN", "CLNACC": "CLNACC"}
+        ),
+        "chr3:5000_G_A": ClinVarVariant(
+            "chr3", 5000, "G", "A", {"CLDN": "CLDN", "CLNACC": "CLNACC"}
+        ),
+    }
+
+    old_clinvar = {
+        "chr1:1000_A_G": ClinVarVariant(
+            "chr1", 1000, "A", "G", {"CLDN": "CLDN", "CLNACC": "CLNACC"}
+        ),
+        "chr2:2000_A_G": ClinVarVariant(
+            "chr2", 2000, "A", "G", {"CLDN": "CLDN", "CLNACC": "CLNACC"}
+        ),
+        "chr2:5000_C_T": ClinVarVariant(
+            "chr2", 5000, "C", "T", {"CLDN": "CLDN", "CLNACC": "CLNACC"}
+        ),
+    }
+
+    final_bed_path = tmp_path / "final.bed"
+
+    bed_test_content = """\
+chr1\t1\t5\told_annot
+chr1\t10\t15\told_annot2
+chr1\t1000\t2000\tdefault_annot
+chr1\t3000\t4000\tgene1
+chr2\t5000\t6000\tdefault_annot
+"""
+
+    final_bed_path.write_text(bed_test_content)
+
+    out_dir = tmp_path
+    padding = 5
+
+    new_to_add, old_to_remove = compare_clinvar(
+        new_clinvar, old_clinvar, final_bed_path, padding, out_dir, keep_tmp=False
+    )
+
+    print(new_to_add)
+    print(old_to_remove)
+
+
+def test_main(tmp_path: Path):
```

Comment:
Yes, I think I initially intended to do as you say and check the outputs a bit as well. I'll look into it

Reply:
OK, it was worth putting some energy here as well, made me found a small output annotation error. Out of the tests I think this might be the most valuable part, and a natural thing to use to test the development.